convert2rhel/checks.py

```diff
@@ -835,3 +841,81 @@ def check_dbus_is_running():
         "Could not find a running DBus Daemon which is needed to register with subscription manager.\n"
         "Please start dbus using `systemctl start dbus` or (on CentOS Linux 6), `service messagebus start`"
     )
+
+
+def _is_initramfs_file_valid(filepath):
+    """Internal function to verify if an initramfs file is corrupted.
+
+    This method will rely on using lsinitrd to do the validation. If the
+    lsinitrd returns other value that is not 0, then it means that the file is
+    probably corrupted or may cause problems during the next reboot.
+
+    :param filepath: The path to the initramfs file.
+    :type filepath: str
+    :return: A boolean to determine if the file is corrupted.
+    :rtype: bool
+    """
+    logger.info("Checking if the '%s' file is not corrupted.", filepath)
```

Comment:
See note for: https://github.com/oamg/convert2rhel/pull/721/files#diff-cf19bc90f65d9e26e1b6fa7dcd84a18dfba097f65706e211a4f4ac71b1b467f6R897 to see if you want to apply this suggestion
```suggestion
    logger.info("Checking if the '%s' file is valid.", filepath)
    
    logger.debug("Checking if the '%s' file exists.", filepath)    
    if not os.path.exists(filepath):
        return False
        
    logger.debug("Checking if the '%s' file is not corrupted.", filepath)
```